### ISSUE TITLE
Open files from yazi in a helix buffer instead of a zellij new pane

### ISSUE THREAD

**Issue by luccahuguet** [luccahuguet/zellij#8]
No idea how to do that, but things I think of, I'll document here

**Comment by adriangalilea** [comment]
This is what I'm looking for

**Comment by luccahuguet** [comment]
> This is what I'm looking for

This should be unblocked once helix's plugin system is first released (in six to 9 months), so someone could create a plugin that enables that 

but at the same time, we should be getting a file tree plugin, so it is not clear whether having a "yazi sidebar" would still be of value... Why would we not use helix's own file tree plugin? 

🤔

**Comment by luccahuguet** [comment]
if this is possible sooner, i shall implement it!

**Comment by adriangalilea** [comment]
https://github.com/user-attachments/assets/a9dc9820-48d7-4f74-9a3a-413c8c7eeeb3
_Running this on a pi zero through ssh, hence why it's struggling._

As seen in [this github issue](https://github.com/zellij-org/zellij/issues/3018#issuecomment-2099408250) you can make it load in the current buffer, you can even select multiple files.

I think the ideal workflow is:

- side-panel that is hid by default.
- <kbd>alt</kbd>+<kbd>f</kbd> to show it
  - <kbd>esc</kbd> to close it
  - <kbd>enter</kbd> to move the selected files into the buffer and hid the side-panel again.
  
That way the file-tree can be inspected easily with a keybind that you can easily close again, so you can use it both for having a view as well as selecting the files.

I believe this is possible to do as of now, but I won't bother since my current workflow is good enough for me, and I rather work that continue tinkering atm :)

**Comment by luccahuguet** [comment]
Awesome! Looks very cool. 

I will probably point to your solution in the readme, I bet it will help many people

I might take a stab at doing this with nushell

thank you for taking the time

**Comment by luccahuguet** [comment]
I give in, let's roll with a bash script

**Comment by luccahuguet** [comment]
this is the dumb version:

open_file.sh
```bash
#!/bin/sh

# Move focus to the next pane
zellij action focus-next-pane
zellij action write 27 # press esc
zellij action write-chars ":open $1"
zellij action write 13 # press enter
```

yazi.toml
```toml
# yazi.toml

[manager]
ratio = [0, 4, 0]

[opener]
# yazelix v4
edit = [
  { exec = '~/.config/zellij/yazi/open_file.sh $1', desc = "Open File in a new pane" },
]

# yazelix v3
# edit = [{ exec = 'zellij edit "$1"', desc = "Open File in a new pane" }]
```

**Comment by luccahuguet** [comment]
the limitation here is that it does not know whether helix is in the next pane or not. 

so if you hit enter on a file in yazi, it will try to open it anyway, it will hit `ESC` then type `:open path_to_file` and hit `ENTER`

in the pane next to yazi

so hopefully it does not cause any accident

**Comment by luccahuguet** [comment]
this is the slightly less dumb version, and it can only check wheter helix is open if helix was opened from pane directly

if it was opened from the sidebar (with `zelli edit` or something then the output of `zellij action list-clients` changes to something else:
```
CLIENT_ID ZELLIJ_PANE_ID RUNNING_COMMAND
1         terminal_2     nu -c zellij action list-clients
```

**Comment by luccahuguet** [comment]
yazelix v4 will probably be shipping with the slight less dumb version unless a truly smart one gets developed... 👀

**Comment by luccahuguet** [comment]
ok, I fixed it. 

list-clients now works on both cases because if helix is not open, it will open helix step by step, first creating the new pane, then typing "hx $1", then sleeping for 0.3 seconds, and pressing enter

this way the `hx` appears in the running_command from list-clients 

```bash
#!/bin/sh

# Move focus to the next pane
zellij action focus-next-pane

# Get the running command in the current pane
RUNNING_COMMAND=$(zellij action list-clients | awk 'NR==2 {print $3}')

# Check if the command running in the current pane is helix (hx)
if echo "$RUNNING_COMMAND" | grep -q "/hx$"; then
    # The current pane is running helix, use zellij actions to open the file
    zellij action write 27
    zellij action write-chars ":open $1"
    zellij action write 13
else
    # The current pane is not running helix, so open helix in a new pane
    zellij action new-pane
    sleep 0.3
    zellij action write-chars "hx $1"
    zellij action write 13

    # warning from before I got it working
    # zellij action new-pane --name "WARNING" -- "echo" "please open helix in a pane right next to the sidebar (to the right of it) using 'hx path_to_file'"
fi
```

**Comment by luccahuguet** [comment]
I should probably merge this tomorrow, possibly in a v4 branch

**Comment by luccahuguet** [comment]
solved by #16
